R/load.r

```diff
@@ -94,11 +94,14 @@ load_all <- function(pkg = ".", reset = TRUE, recompile = FALSE,
   # in the DESCRIPTION file
   pkg$collate <- as.package(pkg$path)$collate
 
+  # Forcing all of the promises for the loaded namespace now will avoid lazy-load
+  # errors when the new package is loaded overtop the old one.
+  #
   # Reloading devtools is a special case. Normally, objects in the
```

Comment:
I kept that comment because for the devtools case we are forcing the promises to avoid the lazy load errors _and_ because it lets us continue to use the functions when devtools is being reloaded.

But I am fine removing it if it seems confusing.